addons/account_bank_statement_import/account_bank_statement_import.py

```diff
@@ -109,80 +102,81 @@ def _check_parsed_data(self, cr, uid, stmts_vals, context=None):
         # If importing into an existing journal, its currency must be the same as the bank statement
         if journal_id:
-            journal_currency_id = self.pool.get('account.journal').browse(cr, uid, journal_id, context=context).currency.id
+            journal_currency_id = self.env['account.journal'].browse(journal_id).currency.id
             if currency_id and currency_id != journal_currency_id:
                 raise UserError(_('The currency of the bank statement is not the same as the currency of the journal !'))
 
         # If there is no journal, create one (and its account)
         if not journal_id and account_number:
-            journal_id = self._create_journal(cr, uid, currency_id, account_number, context=context)
+            journal_id = self._create_journal(currency_id, account_number)
             if bank_account_id:
-                bank_pool.write(cr, uid, [bank_account_id], {'journal_id': journal_id}, context=context)
+                bank_account.write({'journal_id': journal_id})
 
         # If we couldn't find/create a journal, everything is lost
         if not journal_id:
             raise UserError(_('Cannot find in which journal import this statement. Please manually select a journal.'))
-
         return journal_id
 
-    def _create_journal(self, cr, uid, currency_id, account_number, context=None):
+    @api.multi
+    def _create_journal(self, currency_id, account_number):
         """ Create a journal and its account """
-        wmca_pool = self.pool.get('wizard.multi.charts.accounts')
-        company = self.pool.get('res.users').browse(cr, uid, uid, context=context).company_id
+        MultiChartsAccounts = self.env['wizard.multi.charts.accounts']
+        company = self.env.user.company_id
 
         vals_account = {'currency_id': currency_id, 'acc_name': account_number, 'account_type': 'bank', 'currency_id': currency_id}
-        vals_account = wmca_pool._prepare_bank_account(cr, uid, company, vals_account, context=context)
-        account_id = self.pool.get('account.account').create(cr, uid, vals_account, context=context)
+        vals_account = MultiChartsAccounts._prepare_bank_account(company, vals_account)
+        account_id = self.env['account.account'].create(vals_account).id
 
         vals_journal = {'currency_id': currency_id, 'acc_name': _('Bank') + ' ' + account_number, 'account_type': 'bank'}
-        vals_journal = wmca_pool._prepare_bank_journal(cr, uid, company, vals_journal, account_id, context=context)
-        return self.pool.get('account.journal').create(cr, uid, vals_journal, context=context)
+        vals_journal = MultiChartsAccounts._prepare_bank_journal(company, vals_journal, account_id)
+        return self.env['account.journal'].create(vals_journal).id
 
-    def _create_bank_account(self, cr, uid, account_number, journal_id=False, partner_id=False, context=None):
+    @api.multi
+    def _create_bank_account(self, account_number, journal_id=False, partner_id=False):
```

Comment:
Use @api.model instead of @api.multi, where there is no operation on recordset(s).